demos/random_cut_demo/src/main/scala/io/hydrosphere/rrcf/impl/Reservoir.scala

```diff
@@ -21,8 +21,7 @@ class TimeDecayReservoir(var timeDecay: Int, storage: Array[DataPoint]) extends
 
   // Exponentially weight points according to their timestamps
   private def scorePoint(point: TimestampedDataPoint): Double = {
-    // TODO increasing coefficient in the nominator will smooth  exponent?
-    math.pow(r nextDouble(), 1.0 / (timeDecay - point.timestamp))
+    math.pow(r nextDouble(), timeDecay / (timeDecay - point.timestamp))

```

Comment:
1. inconsistent method invocation `r.nextDouble()`
2. is there any specific reason why exponent must be smoothed by `timeDecay`?  is it better to add new parameter variable `decaySmooth` and set it from forest parameters?

Reply:
1. ok
2. By experiments setting numerator to timeDecay gives most diverse trees in forest (65% point overlap)